### PR TITLE
engine: per try idle timeout

### PR DESCRIPTION
**Pull request by junr03**
Description: expose the new per try idle timeout via the engine builder.
Risk Level: low 
Testing: unit test
Docs Changes: added

### CODE REVIEW

**Review by buildbreaker — DISMISSED** [review]

**Comment by junr03** [comment]
https://github.com/envoyproxy/envoy-mobile/pull/1804 is needed for this to go green

File: docs/root/api/starting_envoy.rst [hunk 1]
```diff
@@ -174,7 +174,7 @@ Specify the rate at which Envoy Mobile should flush its queued stats.
 
 Specifies the length of time a stream should wait without a headers or data event before timing out.
 Defaults to 15 seconds.
-See `the Envoy docs <https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-stream-idle-timeout>`_
```
Comment (rebello95) [inline docs/root/api/starting_envoy.rst:177]: Is this deliberate?

File: docs/root/api/starting_envoy.rst [hunk 2]
```diff
@@ -185,6 +185,23 @@ for further information.
   // Swift
   builder.addStreamIdleTimeoutSeconds(5)
 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``addPerTryIdleTimeoutSeconds``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specifies the length of time a retry (including the initial attempt) should wait without a headers
+or data event before timing out. Defaults to 15 seconds.
```
Comment (rebello95) [inline docs/root/api/starting_envoy.rst:193]: Does this apply to trailers?

**Review by rebello95 — APPROVED** [review]

File: docs/root/api/starting_envoy.rst (hunk 2 above)
Comment (junr03) [reply docs/root/api/starting_envoy.rst:193]: Yes, this should say "wait for bytes". That would be more comprehensive.

Comment (junr03) [reply docs/root/api/starting_envoy.rst:193]: @buildbreaker let me fix this in a subsequent PR so we don't have to run CI on this again.

File: docs/root/api/starting_envoy.rst (hunk 1 above)
Comment (junr03) [reply docs/root/api/starting_envoy.rst:177]: not sure why @buildbreaker updated. But still seems to generate correctly.

**Comment by junr03** [comment]
@buildbreaker going to merge so you and @rebello95 can wire up and we can test.

**Comment by rebello95** [comment]
sounds good

**Pull request merged**